bluebrain/repo-patches/packages/neuron/package.py

```diff
@@ -93,20 +93,21 @@ class Neuron(CMakePackage):
     variant("caliper", default=False, description="Add LLNL/Caliper support")
 
     # extra variants from coreneuron recipe
-    variant("gpu", default=False, description="Enable GPU build")
+    variant("gpu", default=False, description="Enable GPU build", when="@9: +coreneuron")
     variant("knl", default=False, description="Enable KNL specific flags")
-    variant("unified", default=False, description="Enable Unified Memory with GPU build")
-    variant("openmp", default=False, description="Enable OpenMP support")
+    variant("unified", default=False, description="Enable Unified Memory with GPU build", when="+gpu")
+    variant("openmp", default=False, description="Enable OpenMP support", when="@9:")
     variant("report", default=True, description="Enable SONATA and binary reports")
     variant("shared", default=True, description="Build shared library")
-    variant("nmodl", default=True, description="Use NMODL instead of MOD2C")
+    variant("nmodl", default=True, description="Use NMODL instead of MOD2C", when="@9:9.0.a6 +coreneuron")
     variant(
         "codegenopt",
         default=False,
         description="Use NMODL with codedgen ionvar copies optimizations",
+        when="+nmodl",
```

Comment:
`+codegenopt` also makes sense for the latest/develop/mod2c-free version, which doesn't have `+nmodl`.

I think `@9:9.0.a6 +coreneuron+nmodl` and `@develop +coreneuron` (I'm assuming `.a6` still supports `mod2c`) should both allow `+codegenopt`.